connectd/peer_exchange_initmsg.c

```diff
@@ -40,12 +40,27 @@ static bool contains_common_chain(struct bitcoin_blkid *chains)
 /* Here in case we need to read another message. */
 static struct io_plan *read_init(struct io_conn *conn, struct early_peer *peer);
 
+struct remote_addr_any {
+	u8 type; /* ipv4:1  ipv6:2 */
+	union {
+		struct {
+			u8 addr[4];
+			u16 port;
+		} ipv4;
+		struct {
+			u8 addr[16];
+			u16 port;
+		} ipv6;
+	} u;
+}__attribute__((packed));
+
```

Comment:
Nope.  You need to demashall properly; this makes unwarrented assumptions about the alignment of the packet, esp. the u16.